onscale_client/api/rest_api.py

```diff
@@ -1406,7 +1406,7 @@ def project_create(
         project_title: str = None,
         project_goal: str = None,
         core_hour_limit: int = None,
-    ) -> datamodel.ProjectCreateResponse:
+    ) -> datamodel.ProjectCreateRequest:
```

Comment:
I don't understand. How can the method `project_create()` return a request instead of a response?

Reply:
The request should remain as a request. I think the data model is missing the ProjectCreateResponse, which should be

```json
{
  "traceId": "string",
  "projectId": "string",
  "accountId": "string",
  "hpcId": "string",
  "userId": "string",
  "projectTitle": "string",
  "projectGoal": "string",
  "createDate": 1585700000000,
  "lastUpdate": 1585700000000,
  "coreHourUsed": 0,
  "designList": [
    {
      "traceId": "string",
      "designId": "design-id",
      "projectId": "string",
      "userId": "string",
      "designTitle": "string",
      "designDescription": "string",
      "designGoal": "string",
      "physics": "MECHANICAL",
      "createDate": 1585700000000,
      "parentDesignId": "string",
      "designInstanceList": [
        {
          "traceId": "string",
          "designInstanceId": "design-instance-id",
          "designId": "design-id",
          "projectId": "string",
          "userId": "string",
          "designInstanceTitle": "Instance Title",
          "description": "string",
          "createDate": 1585700000000,
          "designInstanceHash": "string",
          "parentDesignInstanceId": "string",
          "analysisList": [
            {
              "traceId": "string",
              "analysisId": "analysis-id",
              "projectId": "string",
              "designId": "string",
              "designInstanceId": "string",
              "blobId": "string",
              "accountId": "string",
              "userId": "string",
              "jobId": "string",
              "analysisType": "string",
              "solver": "string",
              "timeDuration": 0,
              "timeStep": 0,
              "createDate": 1585700000000,
              "cipherSalt": "00000000-0000-0000-0000-000000000000",
              "estimateHash": "string",
              "precision": "single",
              "pythonHash": "string",
              "lifeTimeLimit": 0,
              "requiredBlobs": [
                "string"
              ],
              "physicsTypes": [
                "string"
              ]
            }
          ],
          "archived": true,
          "locked": true
        }
      ],
      "archived": true
    }
  ],
  "userIdList": [
    "string"
  ],
  "lastUpdateByMe": 1585700000000,
  "myAccessType": "string",
  "archived": true
}
```

Reply:
maybe this is what we should be using? `class Project(BaseModel)`?

Reply:
Thanks @jeremyonscale 
You were right, I corrected and changed ProjectCreateRequest into Project and then I implemented some function to list the projects in my account to check if it was working and it does!
The changes are in the last commit: https://github.com/OnScale/OnScale-Cloud-Client/pull/13/commits/374c1b0bc2028c7ab4d514b5edc92c2911a0fd50

I think that all the API calls are available in Swagger, but those for the **Project** model were never really implemented.
This **ProjectCreateResponse** was probably a new model is the process of being implemented, but it never was, so it stayed like that.

Reply:
Looks good. Let me try it out.